source/Application/Model/Article.php

```diff
@@ -488,6 +488,10 @@ public function __construct($aParams = null)
                 $this->$sParam = $mValue;
             }
         }
+        
+        // move this and the following dependencies to the constructor and do not use getters inside of methods
+        $this->config = $this->getConfig();
```

Comment:
On top - introduce interfaces for all kinds of dependencies:
```php
interface ConfigurableInterface
{
    public function setConfig(ConfigInterface $config);
}

interface DataDependentInterface
{
    public function setDataProvider(DatabaseProviderInterface $dataProvider);
}
...
```

So, depending on Interfaces implemented, ``oxNew()`` can nicely inject dependencies via setters